target_chains/solana/programs/pyth-solana-receiver/src/lib.rs

```diff
@@ -0,0 +1,260 @@
+                    accounts:   vec![
+                        AccountMeta::new_readonly(ctx.accounts.guardian_set.key(), false),
+                        AccountMeta::new_readonly(ctx.accounts.bridge_config.key(), false),
+                        AccountMeta::new_readonly(ctx.accounts.signature_set.key(), false),
+                        AccountMeta::new(ctx.accounts.vaa.key(), false),
+                        AccountMeta::new(ctx.accounts.payer.key(), true),
+                        AccountMeta::new_readonly(Clock::id(), false),
+                        AccountMeta::new_readonly(Rent::id(), false),
+                        AccountMeta::new_readonly(ctx.accounts.system_program.key(), false),
+                    ],
+                    data:       (POST_VAA, post_vaa_ix_data).try_to_vec()?,
+                };
+                invoke(&post_vaa_ix, ctx.accounts.to_account_infos().as_slice())?;
+            }
+        }
+        Ok(())
+    }
+
+    /// Verify the updates using the posted_vaa account
+    ///  * `vaa_hash` hash post of the post_vaa data to derive the address of the post_vaa account
+    ///  * `emitter_chain` expected emitter_chain from the post_vaa account
+    ///  * `price_updates` Vec of bytes for the updates to verify and post on-chain
+    ///
+    /// TODO:
+    ///    - use a `config` account that can only be modified by governance for checking emitter_chain
+    ///      and other constraints
+    ///
+
+    #[allow(unused_variables)]
+    pub fn post_updates(
+        ctx: Context<PostUpdates>,
+        vaa_hash: [u8; 32], // used for pda seeds
+        emitter_chain: u16,
+        price_updates: Vec<Vec<u8>>,
+    ) -> Result<()> {
+        let vaa = &ctx.accounts.posted_vaa;
+        require_eq!(
+            vaa.emitter_chain(),
+            emitter_chain,
+            ReceiverError::InvalidEmitterChain
+        );
+        let wh_message = WormholeMessage::try_from_bytes(vaa.payload.as_slice())
+            .map_err(|_| ReceiverError::InvalidWormholeMessage)?;
+        msg!("constructed wh_message {:?}", wh_message);
+        let root: MerkleRoot<Keccak160> = MerkleRoot::new(match wh_message.payload {
+            WormholePayload::Merkle(merkle_root) => merkle_root.root,
+        });
+
+        let mut count_updates = 0;
+
+        let price_updates_len = price_updates.len();
+        for price_update in price_updates {
+            let merkle_price_update =
+                from_slice::<byteorder::BE, MerklePriceUpdate>(price_update.as_slice())
+                    .map_err(|_| ReceiverError::DeserializeUpdateFailed)?;
+            let message_vec = Vec::from(merkle_price_update.message);
+            if !root.check(merkle_price_update.proof, &message_vec) {
+                return err!(ReceiverError::InvalidPriceUpdate);
+            }
+            let msg = from_slice::<byteorder::BE, Message>(&message_vec)
+                .map_err(|_| ReceiverError::InvalidAccumulatorMessage)?;
```

Comment:
shouldn't this be a `DeserializedMessageFailed`?

Reply:
updated this thanks